File: benchmarks/primitives/peers/array_set.zig
const std = @import("std");
fn now() f64 {
    var ts: std.c.timespec = undefined;
    _ = std.c.clock_gettime(std.c.CLOCK.MONOTONIC, &ts);
    return @as(f64, @floatFromInt(ts.sec)) + @as(f64, @floatFromInt(ts.nsec)) / 1e9;
}
// Sequential element write over a fixed [1024] stack array, nested reps times.
// The loop-carried chk + final read-back keep the stores alive. Mirrors
// array_set.w.
pub fn main() void {
    var vseed: i64 = 976562;
    const reps: i64 = @as(*volatile i64, &vseed).*;
    var tab: [1024]i64 = [_]i64{0} ** 1024;
    const t0 = now();
    var chk: i64 = reps;
    var r: i64 = 0;
    while (r < reps) : (r += 1) { var k: i64 = 0; while (k < 1024) : (k += 1) { tab[@intCast(k)] = chk ^ k; chk +%= 1; } }
    const el = now() - t0;
    const out: i64 = chk ^ tab[0] ^ tab[1023];
    const ops: i64 = reps * 1024;
    var buf: [256]u8 = undefined;
    const outs = std.fmt.bufPrint(&buf, "{d}\nops: {d}\nelapsed: {d:.6}s\n", .{ out, ops, el }) catch return;
    _ = std.c.write(1, outs.ptr, outs.len);
}

File: benchmarks/primitives/peers/array_set.rs
use std::time::Instant;
// Sequential element write over a fixed [1024] stack array, nested reps times.
// The loop-carried chk + final read-back keep the stores alive. Mirrors
// array_set.w.
fn main(){ let reps:i64=std::hint::black_box(976562); let mut tab=[0i64;1024];
 let t0=Instant::now();
 let mut chk:i64=reps;
 for _r in 0..reps { for k in 0..1024 { tab[k]=chk^(k as i64); chk=chk.wrapping_add(1); } }
 let el=t0.elapsed().as_secs_f64();
 let out=chk^tab[0]^tab[1023];
 let ops=reps*1024;
 println!("{}\nops: {}\nelapsed: {:.6}s",out,ops,el); }
